spec/models/reception_spec.rb

```diff
@@ -37,6 +37,28 @@
     it { is_expected.not_to be_valid }
   end
 
+  context 'with an invalid sample' do
+    let(:attributes) do
+      { plates_attributes: [
+        {
+          type: 'plates',
+          barcode: generate(:barcode),
+          wells_attributes: [
+            {
+              position: 'A1',
+              request: { library_type: 'lib', data_type: 'dt', cost_code: generate(:cost_code), external_study_id: generate(:external_study_id) },
+              sample: { name: generate(:sample_name), species: 'Human', external_id: generate(:uuid), retention_instruction: 'invalid_instruction' }
+            }
+          ]
+        }
+      ], tubes_attributes: [] }
+    end
+
+    it 'raises an ArgumentError' do
+      expect { reception.valid? }.to raise_error(ArgumentError)
```

Comment:
So I don't think this is the validation you wrote causing this, am I right?  If so, then you don't need to validate the value in that model attribute.

Reply:
I think what's causing the `ArgumentError` is the `enum` mapping defined in the model. The intention behind this test is to check the validity of a reception resource when an invalid `sample` is sent into the endpoint.

I didn't quite get what you meant by:

> ... then you don't need to validate the value in that model attribute.

Reply:
What I'm saying is that if you removed the line `validates :retention_instruction, inclusion: { in: VALID_RETENTION_INSTRUCTIONS }, allow_nil: true` from the model, I suspect the tests will still show that `ArgumentError` occurs when you want it to.  I might be wrong.

Reply:
Nope you're correct. I just forgot to remove it. The `enum` mapping does the validation; it doesn't require an additional callback in this case.